### PR TITLE
Feature/portal

### LINKED ISSUE #211: External source should be an enum in the FunderMaps.Core
Take all geom data sources from the database and convert this into a core enum. Cannot be done at this moment because of unstable work

### PR DESCRIPTION
**Pull request by yorickdewid**
**What issue does this PR address?**
Expose unauthorized URL `api/incident-portal`. URI is to be taken as endpoint in the FunderMaps Incident Portal.

**Does this PR introduce a breaking change?**
Nope

**Other information**:
Fixed #211 in same PR

### CODE REVIEW

File: src/FunderMaps.WebApi/Controllers/Portal/IncidentPortalController.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        {
+            _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+            _incidentUseCase = incidentUseCase ?? throw new ArgumentNullException(nameof(incidentUseCase));
+            _geocoderUseCase = geocoderUseCase ?? throw new ArgumentNullException(nameof(geocoderUseCase));
+        }
+
+        /// <summary>
+        ///     Upload document to the backstore.
+        /// </summary>
+        /// <param name="input">See <see cref="IFormFile"/>.</param>
```
Comment (tabeckers) [inline src/FunderMaps.WebApi/Controllers/Portal/IncidentPortalController.cs:42]: For referencing another file, will we use the standard to add the word "See"?

File: src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs [hunk 2]
```diff
@@ -91,41 +96,24 @@ public async Task<IActionResult> GetAllAsync([FromQuery] PaginationModel paginat
...
             {
-                incident.Meta = new
-                {
-                    UserAgent = Request.Headers["User-Agent"].ToString(),
-                    RemoteAddress = HttpContext.Connection.RemoteIpAddress?.ToString(),
-                    Gateway = Constants.IncidentGateway,
-                };
-            }
+                SessionUser = sessionUser.Id,
+                sessionOrganization = sessionOrganization.Id,
```
Comment (tabeckers) [inline src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs:114]: Should also be PascalCase for consistency

File: src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs [hunk 3]
```diff
@@ -91,41 +96,24 @@ public async Task<IActionResult> GetAllAsync([FromQuery] PaginationModel paginat
...
-                    SessionUser = sessionUser.Id,
-                    sessionOrganization = sessionOrganization.Id,
-                    Gateway = Constants.IncidentGateway,
-                };
-            }
-            else
+            var sessionUser = await _authManager.GetUserAsync(User);
+            var sessionOrganization = await _authManager.GetOrganizationAsync(User);
+
+            incident.Meta = new
```
Comment (tabeckers) [inline src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs:111]: Is it a good idea to store this as a json object of any format? Shouldn't we be a bit more consistent with this?

File: src/FunderMaps.WebApi/DataTransferObjects/AddressDto.cs [hunk 4]
```diff
@@ -30,23 +31,5 @@ public sealed class AddressDto
         /// </summary>
         [Required(AllowEmptyStrings = false)]
         public string Street { get; set; }
-
```
Comment (tabeckers) [inline src/FunderMaps.WebApi/DataTransferObjects/AddressDto.cs:33]: Why was this removed?

File: tests/FunderMaps.IntegrationTests/Backend/Portal/IncidentPortalTests.cs [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+        [InlineData("1111YY")]
+        public async Task GetAllAddressByQueryReturnEmptyList(string query)
+        {
+            // Act
+            var response = await _client.GetAsync($"api/incident-portal/address-suggest?query={query}");
+            var returnList = await response.Content.ReadFromJsonAsync<List<Address>>();
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal(0, returnList.Count);
```
Comment (tabeckers) [inline tests/FunderMaps.IntegrationTests/Backend/Portal/IncidentPortalTests.cs:225]: Use Assert.Empty()

**Review by tabeckers — CHANGES_REQUESTED** [review]
Looked into all files and left some comments in the files themselves.

File: src/FunderMaps.WebApi/Controllers/Portal/IncidentPortalController.cs (hunk 1 above)
Comment (yorickdewid) [reply src/FunderMaps.WebApi/Controllers/Portal/IncidentPortalController.cs:42]: I've always done so

File: src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs (hunk 3 above)
Comment (yorickdewid) [reply src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs:111]: There is no mention of JSON in the controller, could have been XML/TOML/CONF

File: .vscode/default.code-workspace [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
```
[the remark this answers is not in the document]
Comment (yorickdewid) [reply .vscode/default.code-workspace:1]: VS Workspace is not bound to specific machine

File: src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs (hunk 2 above)
Comment (yorickdewid) [reply src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs:114]: Nope, because its a local var.
image

File: src/FunderMaps.WebApi/DataTransferObjects/AddressDto.cs (hunk 4 above)
Comment (yorickdewid) [reply src/FunderMaps.WebApi/DataTransferObjects/AddressDto.cs:33]: It exposes internals to unauthorized users

File: tests/FunderMaps.IntegrationTests/Backend/Portal/IncidentPortalTests.cs (hunk 5 above)
Comment (yorickdewid) [reply tests/FunderMaps.IntegrationTests/Backend/Portal/IncidentPortalTests.cs:225]: Done

File: src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs (hunk 3 above)
Comment (tabeckers) [reply src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs:111]: Regardless of the format, do we really want an anonymous object here?

File: src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs (hunk 2 above)
Comment (tabeckers) [reply src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs:114]: I mean it like `SessionOrganization = sessionOrganization.Id,`

File: src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs (hunk 3 above)
Comment (yorickdewid) [reply src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs:111]: No do don't. Can use a local type

File: src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs (hunk 2 above)
Comment (yorickdewid) [reply src/FunderMaps.WebApi/Controllers/Report/IncidentController.cs:114]: I only now see what you mean. There is a missing capital. ofc. Fixing it

**Review by tabeckers — APPROVED** [review]
Done
